src/main/java/org/opensearch/sdk/SDKHttpRequest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.http.HttpResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class helps to get instance of RestRequest
+ */
+public class SDKHttpRequest implements HttpRequest {
+    private final RestRequest.Method method;
+    private final String uri;
+    private final BytesReference content;
+    private final Map<String, List<String>> headers;
+
+    /**
+    * Instantiates this class with a copy of request
+    *
+    * @param request The request
+    */
+    public SDKHttpRequest(ExtensionRestRequest request) {
+        this.method = request.method();
+        this.content = request.content();
+        this.headers = request.headers();
+
+        Path path = Paths.get(request.path());
+        this.uri = path.toUri().toString();
+    }
+
+    /**
+    * This method returns request method
+    * @return A method of request
+    */
+    @Override
+    public RestRequest.Method method() {
+        return method;
+    }
+
+    /**
+    * This method returns request uri
+    * @return URI of request
+    */
+    @Override
+    public String uri() {
+        return uri;
+    }
+
+    /**
+    * This method returns request content
+    * @return content of request
+    */
+    @Override
+    public BytesReference content() {
+        return content;
+    }
+
+    /**
+    * This method returns request headers
+    * @return map containing headers
+    */
+    @Override
+    public Map<String, List<String>> getHeaders() {
+        return headers;
+    }
+
+    /**
+    * This method returns request cookies
+    * @return list containing cookies
+    */
+    @Override
+    public List<String> strictCookies() {
+        return null;
+    }
+
+    /**
+    * This method returns request HTTP protocol version
+    * @return version of request HTTP protocol
+    */
+    @Override
+    public HttpVersion protocolVersion() {
+        return null;
```

Comment:
Since you are changing the request to add headers, and I've asked to pass uri instead of path, let's also add passing this enum in the request and returning it here.